superduper/base/metadata.py

```diff
@@ -229,59 +231,92 @@ class MetaDataStore:
     """
     Abstraction for storing meta-data separately from primary data.
 
-    :param db: Datalayer instance.
-    :param cache: Cache instance.
+    :param db: Datalayer instance for saving components.
+    :param parent_db: Parent Datalayer instance for saving primary data.
     """
 
-    def __init__(self, db, cache: Cache | None = None):
+    def __init__(self, db: 'Datalayer', parent_db: 'Datalayer'):
         self.db = db
-        self.cache = cache
+        self.parent_db = parent_db
+        self._schema_cache: t.Dict[str, Schema] = {}
+
+    def __getitem__(self, item: str):
+        return self.db[item]
+
+    def init(self):
+        """Initialize the metadata store."""
+        self.db.databackend.create_table_and_schema(
+            'Table',
+            Table.class_schema,
+            primary_id='uuid',
+        )
+
+        try:
+            r = self.db['Table'].get(identifier='Table')
+        except NonExistentMetadataError:
```

Comment:
If I 'm not mistaken, the Exception does not come from `.get`. 

Instead, `.get` returns None, and because of that you need to raise a MetadataNotExists exception. 

Like:

```
        r = self.db[component].get(identifier=identifier, version=version, raw=True)

        if r is None:
            raise NonExistentMetadataError(
                f'Object {identifier} does not exist in metadata for {component}'
            )
```

Reply:
The get method here isn’t the one from dict; it’s implemented in the Query class.